pkg/alert/alert.go

```diff
@@ -0,0 +1,427 @@
+// Copyright 2019 Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package alert
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/gardener/test-infra/pkg/util/slack"
+	"github.com/go-logr/logr"
+	"github.com/olekukonko/tablewriter"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"sort"
+	"strings"
+	"time"
+)
+
+type Alert struct {
+	log logr.Logger
+	cfg Config
+	ctx context.Context
+}
+
+//TestDetails describes a test
+type TestDetails struct {
```

Comment:
what do you mean with `describes a test`? Is one test dimension to be alerted?
Also is this the data structure that is persisted in es?

Reply:
this describes a test which is used to generate slack result table for alert and recover messages. Enhanced comment to `//TestDetails describes a test which is used for alert message` is this helpful?